### PR TITLE
Add timeout to EdgeAgent - iotedged operations, and add more logs

### PR DESCRIPTION
**Pull request by varunpuranik**

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs [hunk 1]
```diff
@@ -194,7 +200,8 @@ protected override void HandleException(Exception exception, string operation)
                     }
 
                 default:
-                    throw exception;
+                    ExceptionDispatchInfo.Capture(exception).Throw();
```
Comment (arsing) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs]: There are a dozen or so more cases of ` throw (?!new)` under `edge-agent/` and `edge-hub/`. I guess we should fix those too?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2019_01_30/ModuleManagementHttpClient.cs [hunk 2]
```diff
@@ -22,7 +23,12 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Edgelet.Version_2019_01_30
     class ModuleManagementHttpClient : ModuleManagementHttpClientVersioned
     {
         public ModuleManagementHttpClient(Uri managementUri)
-            : base(managementUri, ApiVersion.Version20190130, new ErrorDetectionStrategy())
+            : this(managementUri, Option.None<TimeSpan>())
+        {
+        }
+
+        internal ModuleManagementHttpClient(Uri managementUri, Option<TimeSpan> timeSpan)
```
Comment (arsing) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2019_01_30/ModuleManagementHttpClient.cs]: `s/Option<TimeSpan> timeSpan/Option<TimeSpan> operationTimeout/`

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs [hunk 3]
```diff
@@ -22,7 +23,12 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Edgelet.Version_2018_06_28
     class ModuleManagementHttpClient : ModuleManagementHttpClientVersioned
     {
         public ModuleManagementHttpClient(Uri managementUri)
-            : base(managementUri, ApiVersion.Version20180628, new ErrorDetectionStrategy())
+            : this(managementUri, Option.None<TimeSpan>())
+        {
+        }
+
+        internal ModuleManagementHttpClient(Uri managementUri, Option<TimeSpan> timeSpan)
```
Comment (arsing) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs]: `s/Option<TimeSpan> timeSpan/Option<TimeSpan> operationTimeout/`

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs (hunk 1 above)
Comment (varunpuranik) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs]: Sure, but this I want to merge to 1.0.7 and for that I want to keep the change smaller. I will send out a follow up with the other places fixed.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2019_01_30/ModuleManagementHttpClient.cs (hunk 2 above)
Comment (varunpuranik) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2019_01_30/ModuleManagementHttpClient.cs]: Sure, will fix

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs (hunk 1 above)
Comment (arsing) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs]: Yeah, I didn't mean to fix them in *this* PR.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs (hunk 3 above)
Comment (varunpuranik) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2018_06_28/ModuleManagementHttpClient.cs]: Fixed

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2019_01_30/ModuleManagementHttpClient.cs (hunk 2 above)
Comment (varunpuranik) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2019_01_30/ModuleManagementHttpClient.cs]: fixed

**Pull request merged**